Hospital/HospitalApi/Controllers/RoomRenovationController.cs

```diff
@@ -0,0 +1,96 @@
+﻿using Hospital.GraphicalEditor.Repository;
+using Hospital.GraphicalEditor.Service;
+using Hospital.RoomsAndEquipment.Model;
+using Hospital.RoomsAndEquipment.Repository;
+using Hospital.Schedule.Service;
+using Hospital.SharedModel.Model.Wrappers;
+using Hospital.SharedModel.Repository.Base;
+using HospitalApi.DTOs;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace HospitalApi.Controllers
+{
+    [Route("api/[controller]/[action]")]
+    [ApiController]
+    public class RoomRenovationController : ControllerBase
+    {
+        private readonly IUnitOfWork _uow;
+
+        public RoomRenovationController(IUnitOfWork uow)
+        {
+            _uow = uow;
+        }
+
+        [HttpGet]
+        public IActionResult GetSurroundingRoomsForRoom([FromQuery(Name = "roomId")] int roomId)
+        {
+            var surroundingRoomsService = new FindingSurroundingRoomsService(_uow);
+            var roomPosition = _uow.GetRepository<IRoomPositionReadRepository>().GetByRoom(roomId);
+            if (roomId <= 0)
+            {
+
+                return BadRequest();
+            }
```

Comment:
You should first validate the roomId and then get the room. Also, what if roomId is valid but no room with that id was found?